opam-ci-check/bin/main.ml

```diff
@@ -16,12 +18,68 @@ let to_exit_code : (unit, string) result Term.t -> Cmd.Exit.code Term.t =
       Printf.eprintf "%s%!" msg;
       1
 
+let download_package ~pkg ~url f =
+  Dir_helpers.with_temp_dir "lint-dune-project-version-" @@ fun dir ->
+  let res =
+    OpamProcess.Job.run
+    @@ OpamRepository.pull_tree
+         (OpamPackage.to_string pkg)
+         (OpamFilename.Dir.of_string dir)
+         (OpamFile.URL.checksum url)
+         [ OpamFile.URL.url url ]
+  in
+  let dir =
+    match res with
+    | OpamTypes.Not_available (_, msg) ->
+        print_endline msg;
+        None
+    | Up_to_date _ | Result _ -> Some dir
+  in
+  f dir
```

Comment:
Yes, I did go with the implementation you suggested before, but I wanted to keep the old behavior through this API change. Don't entirely skip running lint checks if the package source couldn't be downloaded. Only the dune constraints check was being skipped. 

We could discuss the pros and cons and decide to change the behavior if we think completely skipping lints is better in those cases. Maybe it's a way to validate that the source URL is a valid one, even in CI? And we just re-run the lint in case of network errors?